File: webapp/lib/models/delivery.dart
import 'package:latlong2/latlong.dart';

class Delivery {
  late String id;
  late String address;
  late String city;
  late String? postcode;
  late String? customer;
  late String? phone;
  late String restaurant;

  late double? amount;

  late DateTime initTime;
  late DateTime? finishTime;
  late String? dealer;

  late LatLng? coordinates;

  Delivery(
      {required this.id,
      required this.address,
      required this.city,
      required this.restaurant,
      required this.initTime,
      this.postcode,
      this.customer,
      this.phone,
      this.dealer,
      this.amount = 0.0,
      this.finishTime,
      this.coordinates});

  Delivery.fromJson(var json) {
    id = json['_id'];
    address = json['address'];
    restaurant = json['restaurant'];
    city = json['city'];
    initTime = DateTime.parse(json['initTime']);
    amount = (json['amount'] ?? 0.0);
    customer = (json['customer'] ?? '');
    dealer = (json['dealer'] ?? '');
    phone = (json['phone'] ?? '');
    postcode = (json['postcode'] ?? '');
    if (json['longitude'] != null && json['latitude'] != null) {
      coordinates = LatLng(json['latitude'], json['longitude']);
    }
  }
}

File: app/lib/models/delivery.js
export default class Delivery {
  constructor(id, uploadUser, restaurant, brand, dealer, address, city, postcode, initTime, finishTime, amount, customer, phone, latitude, longitude) {
    this.id = id;
    this.uploadUser = uploadUser;
    this.restaurant = restaurant;
    this.brand = brand;
    this.dealer = dealer;
    this.address = address;
    this.city = city;
    this.postcode = postcode;
    this.initTime = initTime;
    this.finishTime = finishTime;
    this.amount = amount;
    this.customer = customer;
    this.phone = phone;
    this.latitude = latitude;
    this.longitude = longitude;
  }
}
   